server/controllers/finance/reports/income_expense/index.js

```diff
@@ -89,10 +89,29 @@ function document(req, res, next) {
 
       const profits = [];
       const losses = [];
+      const totals = {
+        income : { balance : 0, previousBalance : 0 },
+        expense : { balance : 0, previousBalance : 0 },
+      };
 
       tree.walk(node => profits.push(node), true, income);
+      // sum income
+      tree.walk(node => {
+        totals.income.balance += Math.abs(node.balance);
+        totals.income.previousBalance += Math.abs(node.previousBalance);
+      }, true, income);
 
       tree.walk(node => losses.push(node), true, expense);
+      // sum expences
+      tree.walk(node => {
+        totals.expense.balance += Math.abs(node.balance);
+        totals.expense.previousBalance += Math.abs(node.previousBalance);
+      }, true, expense);
+
+      // calculate totals and profit
+      totals.income.difference = Math.abs(totals.income.balance - totals.income.previousBalance);
+      totals.expense.difference = Math.abs(totals.expense.balance - totals.expense.previousBalance);
+      totals.profits = Math.abs(totals.expense.balance - totals.income.balance);
```

Comment:
A better name for this is `totals.net` or `totals.result` because it could be negative.  You usually only call positive numbers "profit" and negative numbers "loss" (which is why this report is called Profit & Loss).